csm/core/services/file_transfer.py

```diff
@@ -89,7 +89,21 @@ def get_file_path(self) -> str:
         return path_to_cached_file
 
     def save_file(self, dir_to_save, filename, overwrite=False):
-        Log.debug(f"Saving f{filename} at f{dir_to_save}")
+        if not os.path.exists(dir_to_save):
+            Log.warn(f"{dir_to_save} not found. Recreating storage directory." )
+            try:
+                original_mask = os.umask(0o007)
```

Comment:
If it is required just for debugging then move it to Log.debug line.

Reply:
It was purposely added to Log.warn because it will printed in logs even if Log level is INFO. Also it will clearly state that folder is missing on location so recreating it.